File: test/new_lang/test4/TestProgram.cs
using static SplashKitSDK.SplashKit;
using static System.Console;

public class Program
{
  public static void Main()
  {
    int i;
    uint ui;
    float f;
    double d;
    short sh;
    ushort ush;
    long l;
    char ch;
    byte uch;
    bool b;

    i = 10;
    f = 1.11f;
    d = 2.22;
    sh = -3;
    ush = 4;
    l = 5;
    ch = 'a';
    uch = (byte)'B';
    b = true;
    ui = 50;


    WriteLine("{0} get {1} updated int {2}", i, GetAndUpdateInt(ref i), i);

    Write("{0} ", i);
    UpdateInt(ref i);
    WriteLine("Updated int: {0}", i);

    Write("{0} ", ui);
    UpdateUint(ref ui);
    WriteLine("Updated uint: {0}", ui);

    Write("{0} ", sh);
    UpdateShort(ref sh);
    WriteLine("Updated short int: {0}", sh);

    Write("{0} ", ush);
    UpdateUshort(ref ush);
    WriteLine("Updated short int: {0}", ush);

    Write("{0} ", f);
    UpdateFloat(ref f);
    WriteLine("Updated float: {0}", f);

    Write("{0} ", d);
    UpdateDouble(ref d);
    WriteLine("Updated double: {0}", d);

    Write("{0} ", l);
    UpdateLong(ref l);
    WriteLine("Updated long: {0}", l);

    Write("{0} ", ch);
    UpdateChar(ref ch);
    WriteLine("Updated char: {0}", ch);

    Write("{0} ", uch);
    UpdateUchar(ref uch);
    WriteLine("Updated char: {0}", uch);

    Write("{0} ", b);
    UpdateBool(ref b);
    WriteLine("Updated bool: {0}", b);
  }
}

File: test/new_lang/test4/test_program.rs
#[path = "../../out/rust/splashkit.rs"]
mod splashkit;
use splashkit::*;

fn main() {
  
  let mut i : i32 = 10;
  let mut f : f32 = 1.11;
  let mut d : f64 = 2.22;
  let mut sh : i16 = -3;
  let mut ush : u16 = 4;
  let mut l : i64 = 5;
  let mut ch : char = 'a';
  let mut uch : u8 = 66;
  let mut b : bool = true;
  let mut ui : u32 = 50;

  println!("i is {}", i);
  println!("get and update {} now {}", get_and_update_int(&mut i), i);

  println!("i is {}", i);
  update_int(&mut i);
  println!("Updated int: {}", i);

  println!("ui is {}", ui);
  update_uint(&mut ui);
  println!("Updated ui: {}", ui);

  println!("f is {}", f);
  update_float(&mut f);
  println!("Updated ui: {}", f);

  // print!(ui, end=" ")
  // update_uint(ui)
  // print!("Updated uint: ", ui)

  // print!(sh, end=" ")
  // update_short(sh)
  // print!("Updated short int: ", sh)

  // print!(ush, end=" ")
  // update_ushort(ush)
  // print!("Updated short int: ", ush)

  // print!(f, end=" ")
  // update_float(f)
  // print!("Updated float: ", f)

  // print!(d, end=" ")
  // update_double(d)
  // print!("Updated double: ", d)

  // print!(l, end=" ")
  // update_long(l)
  // print!("Updated long: ", l)

  // print!(ch, end=" ")
  // update_char(ch)
  // print!("Updated char: ", ch)

  // print!(uch, end=" ")
  // update_uchar(uch)
  // print!("Updated char: ", uch)

  // print!(b, end=" ")
  // update_bool(b)
  // print!("Updated bool: ", b)
}